R/conditions.R

```diff
@@ -243,6 +246,15 @@ stop_unimplemented <- function(x, method) {
   )
 }
 
+stop_scalar_type <- function(x, arg = NULL) {
+  if (is_null(arg)) {
```

Comment:
Are you sure we shouldn't require this argument?

Reply:
I think it would be difficult to make it a requirement. I have two other PRs pending locally and IIRC I call `stop_scalar_type()` without `arg` because passing the arg would require adding it as argument to many vctrs primitives. I'll think about it though.